### PR TITLE
Automate Java cudf jar build with statically linked dependencies

### PR DESCRIPTION
**Pull request by gerashegalov**
This PR automates cudf jar build via build.sh . It uses java CI centos7 Docker image to build a portable jar from the source checked out on the host. The Docker container is run as a normal non-root user https://docs.docker.com/engine/install/linux-postinstall/

Examples:
```bash
# clean, build libcudf, and use `mvn package` with default parallelism
./build.sh clean cudfjar

# build libcudf, and use `mvn install` 
PARALLEL_LEVEL=4  MVN_PHASES="install" ./build.sh cudfjar
```

2. allow overriding GPU architectures with ALL, NATIVE, some subset.

3. Drop GDS from the image
4. Adds parallelism to the libcudfjni build, removes `make` hardcoding
5. Fix: don't disregard `CUDF_CPP_BUILD_DIR`  passed to Maven
6. Use a dedicated CUDF_CPP_BUILD_DIR value to avoid corrupting interactions with `build.sh libcudf`

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -17,12 +17,13 @@ ARGS=$*
...
 REPODIR=$(cd $(dirname $0); pwd)
 
-VALIDARGS="clean libcudf cudf dask_cudf benchmarks tests libcudf_kafka cudf_kafka custreamz -v -g -n -l --allgpuarch --disable_nvtx --show_depr_warn --ptds -h --build_metrics --incl_cache_stats"
+VALIDARGS="clean libcudf cudf cudfjar dask_cudf benchmarks tests libcudf_kafka cudf_kafka custreamz -v -g -n -l --allgpuarch --disable_nvtx --show_depr_warn --ptds -h --build_metrics --incl_cache_stats"
 HELP="$0 [clean] [libcudf] [cudf] [dask_cudf] [benchmarks] [tests] [libcudf_kafka] [cudf_kafka] [custreamz] [-v] [-g] [-n] [-h] [--cmake-args=\\\"<args>\\\"]
    clean                         - remove all existing build artifacts and configuration (start
                                    over)
    libcudf                       - build the cudf C++ code only
    cudf                          - build the cudf Python package
+   cudfjar                       - build cudf JAR with static libcudf using devtoolset-9 toolchain
```
Comment (jlowe) [inline build.sh:26]: Suggest dropping the version here. It's very likely to be left outdated when we update to the next devtoolset version and arguably is too much detail here.
```suggestion
   cudfjar                       - build cudf JAR with static libcudf using devtoolset toolchain
```

File: java/ci/Dockerfile.centos7 [hunk 2]
```diff
@@ -33,16 +33,8 @@ RUN yum install -y git zlib-devel maven tar wget patch ninja-build
 ## pre-create the CMAKE_INSTALL_PREFIX folder, set writable by any user for Jenkins
```
Comment (jlowe) [inline java/ci/Dockerfile.centos7:33]: 2022 copyrights

File: java/ci/Dockerfile.centos7 [hunk 3]
```diff
@@ -33,16 +33,8 @@ RUN yum install -y git zlib-devel maven tar wget patch ninja-build
...
 RUN mkdir /usr/local/rapids && mkdir /rapids && chmod 777 /usr/local/rapids && chmod 777 /rapids
 
-ARG CMAKE_VERSION=3.20.5
+ARG CMAKE_VERSION=3.22.3
 RUN cd /usr/local/ && wget --quiet https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz && \
    tar zxf cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz && \
    rm cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz
 ENV PATH /usr/local/cmake-${CMAKE_VERSION}-linux-x86_64/bin:$PATH
-
-# get GDS user-space lib
```
Comment (jlowe) [inline java/ci/Dockerfile.centos7:42]: Have you tested building in the Docker image with `-DUSE_GDS=ON` as one of the Maven arguments?  The build does not try to use GDS by default.

Comment (gerashegalov) [reply java/ci/Dockerfile.centos7:42]: yes, `USE_GDS=ON` is hardcoded by this PR in build.sh and the build and unit tests are successful.

**Review by jlowe — APPROVED** [review]

**Comment by gerashegalov** [comment]
Made local host's .m2 customizable via `LOCAL_MAVEN_REPO` environment. defaulting to the standard location `~/.m2`

**Review by nvdbaranec — APPROVED** [review]

**Comment by gerashegalov** [comment]
@gpucibot merge

**Pull request merged**